edx_django_utils/monitoring/internal/utils.py

```diff
@@ -96,3 +96,16 @@ def record_exception():
     """
     if newrelic:  # pragma: no cover
         newrelic.agent.record_exception()
+
+
+def background_task(*args, **kwargs):
+    """
+    Handles monitoring for background tasks that are not passed in through the web server like
+    celery and event consuming tasks.
+
+    For more details, see:
+    https://docs.newrelic.com/docs/apm/agents/python-agent/supported-features/monitor-non-web-scripts-worker-processes-tasks-functions
+
+    """
+    if newrelic:  # pragma: no cover
+        newrelic.agent.background_task(*args, **kwargs)
```

Comment:
Here's my best guess, untested...

```suggestion
    if newrelic:  # pragma: no cover
        return newrelic.agent.background_task(*args, **kwargs)
    else:
        return lambda f: f
```

Reply:
Yup, I was queueing up the change, and I think you're right.

Reply:
Hah! I'll be pleased if I was, but not surprised if I wasn't. :-P Despite being pretty comfortable with functional programming, decorators always somehow throw me for a loop. (Becca linked to it as a puzzle.)

Reply:
The parts where it gets translated from functional programming to syntactic sugar always trips me up, too.